frontend/app/routes/protected/person-case/first-name.tsx

```diff
@@ -1,49 +1,123 @@
-import { Form } from 'react-router';
-import type { RouteHandle } from 'react-router';
+import { useMemo } from 'react';
+
+import { data, redirect, useFetcher } from 'react-router';
+import type { ActionFunctionArgs, RouteHandle } from 'react-router';
 
 import { useTranslation } from 'react-i18next';
+import * as v from 'valibot';
+
+import type { Route, Info } from './+types/first-name';
 
+import { commitSession, getSession } from '~/.server/environment/session';
+import { requireAuth } from '~/.server/utils/auth-utils';
+import { firstNameSchema } from '~/.server/validation';
 import { Button } from '~/components/button';
+import { ButtonLink } from '~/components/button-link';
+import { ErrorSummary } from '~/components/error-summary';
 import { InputField } from '~/components/input-field';
-import { Menu, MenuItem } from '~/components/menu';
 import { PageTitle } from '~/components/page-title';
+import { getFixedT } from '~/i18n-config.server';
+import { i18nRoutes } from '~/i18n-routes';
 import { handle as parentHandle } from '~/routes/protected/layout';
+import { getLanguage } from '~/utils/i18n-utils';
+import { getPathByFile } from '~/utils/route-utils';
 
 export const handle = {
   i18nNamespace: [...parentHandle.i18nNamespace, 'protected'],
 } as const satisfies RouteHandle;
 
-export default function FirstName() {
+export async function loader({ context, request }: Route.LoaderArgs) {
+  requireAuth(context.session, new URL(request.url), ['user']);
+  const t = await getFixedT(request, handle.i18nNamespace);
+
+  const session = await getSession(request.headers.get('cookie'));
+
+  return {
+    documentTitle: t('protected:person-case.page-title'),
+    sessionData: session.data,
+  };
+}
+
+export function meta({ data }: { data: { documentTitle: string } }) {
+  return [{ title: data.documentTitle }];
+}
+
+export async function action({ context, request }: ActionFunctionArgs) {
+  const formData = await request.formData();
+  const firstName = formData.get('first-name');
+  // Validation schema
+  const schema = v.object({
+    firstName: firstNameSchema(),
+  });
+
+  // Form data input
+  const input = {
+    firstName: formData.get('first-name'),
+  };
+
+  // Safe-parse form data input
+  const lang = getLanguage(request);
+  const parsedDataResult = v.safeParse(schema, input, { lang });
+
+  if (!parsedDataResult.success) {
+    return data({ errors: v.flatten<typeof schema>(parsedDataResult.issues).nested }, { status: 400 });
+  }
+  // If the first name is valid, store it in the session and redirect to the next page
+
+  const session = await getSession();
+  session.set('firstName', firstName);
+
+  return redirect(getPathByFile('routes/protected/request.tsx', i18nRoutes, request), {
+    headers: {
+      'Set-Cookie': await commitSession(session),
+    },
```

Comment:
Since we're not using cookie-backed sessions, you don't have to send headers or call `commitSession()`.